### DISCUSSION TITLE
Feature Request: Support for Displaying Complex Fields

### DISCUSSION THREAD

**Discussion by blaisewang** [Kong/kong-manager#239] [Ideas]
Hello Kong team,

I am new to Kong and its ecosystem, so please correct me if I am misunderstanding something.

I am currently developing a plugin for Kong and supporting the configuration of multiple servers within my plugin. My plugin schema defines this field as follows:

```lua
{
  servers = {
    type = "array",
    required = true,
    elements = {
      type = "record",
      fields = {
        { host = { type = "string", required = true, }},
        { port = { type = "number", required = true, }},
        { weight = { type = "number", required = false, default = 1, }},
      }
    }
  }
}
```

However, when I view the Servers field in Kong Manager, it only shows [object Object] as depicted in the screenshot below.

<img width="796" alt="image" src="https://github.com/Kong/kong-manager/assets/9657268/c4832cbb-3053-4a33-bdfb-af05bd942cfa">

I would like to confirm whether this is a limitation of Kong Manager or if there is an issue with my field definition. If it is a limitation, I hope that Kong Manager can support the display of such complex fields to make it easier to configure and manage plugins.

Best regards,
Blaise

**Comment by blaisewang** [comment]
Update: I found that the Kong bundled plugin AI Prompt Template has a similar configuration schema. Please refer to `config.templates` in the documentation: https://docs.konghq.com/hub/kong-inc/ai-prompt-template/configuration/#config-templates. 

Kong Manager can display and modify this configuration correctly. I would like to ask if Kong Manager has special compatibility for this plugin? Does a third-party plugin require additional configuration to be displayed correctly?

**Comment by Leopoldthecoder** [comment]
Hi @blaisewang, in Kong Manager 3.7 and prior, we only support this type of schema in official plugins, and that's why `AI Prompt Template` works.

We are now bringing support for record arrays to custom plugins, so I believe in Kong Gateway 3.8 (our next minor release) your plugin should be working correctly. If you are interested, you can check out [Konnect](https://cloud.konghq.com/) (our cloud platform) where the support is already added: 
<img width="1407" alt="image" src="https://github.com/Kong/kong-manager/assets/10095631/e929e1a4-b7c8-4d86-ae63-6973054a7c74">

**Reply by blaisewang** [reply]
@Leopoldthecoder Thank you for your reply. I look forward to the next release.
